fv3core/stencils/dyn_core.py

```diff
@@ -179,381 +197,415 @@ def dyncore_temporaries(shape):
+                    akap,
+                    state.cappa,
+                    state.ptop,
+                    state.phis,
+                    state.omga,
+                    state.ptc,
+                    state.q_con,
+                    state.delpc,
+                    state.gz,
+                    state.pkc,
+                    state.ws3,
+                )
+
+            self._p_grad_c(
+                grid.rdxc,
+                grid.rdyc,
+                state.uc,
+                state.vc,
+                state.delpc,
                 state.pkc,
                 state.gz,
-                state.pk3,
-                state.delp,
-                dt,
-                state.ptop,
-                akap,
+                dt2,
             )
-
-        if spec.namelist.rf_fast:
-            # TODO: Pass through ks, or remove, inconsistent representation vs Fortran.
-            ray_fast.compute(
+            if self.do_halo_exchange:
+                reqc_vector = self.comm.start_vector_halo_update(
+                    state.uc_quantity, state.vc_quantity, n_points=utils.halo
+                )
+                if self.namelist.nord > 0:
+                    reqs["divgd_quantity"].wait()
+                reqc_vector.wait()
+            # use the computed c-grid winds to evolve the d-grid winds forward
+            # by 1 timestep
+            d_sw.compute(
+                state.vt,
+                state.delp,
+                state.ptc,
+                state.pt,
                 state.u,
                 state.v,
                 state.w,
-                state.dp_ref,
-                state.pfull,
+                state.uc,
+                state.vc,
+                state.ua,
+                state.va,
+                state.divgd,
+                state.mfxd,
+                state.mfyd,
+                state.cxd,
+                state.cyd,
+                state.crx,
+                state.cry,
+                state.xfx,
+                state.yfx,
+                state.q_con,
+                state.zh,
+                state.heat_source,
+                state.diss_estd,
                 dt,
-                state.ptop,
-                state.ks,
             )
+            # note that uc and vc are not needed at all past this point.
+            # they will be re-computed from scratch on the next acoustic timestep.
+
+            if self.do_halo_exchange:
+                for halovar in ["delp_quantity", "pt_quantity", "q_con_quantity"]:
+                    self.comm.halo_update(
+                        state.__getattribute__(halovar), n_points=utils.halo
+                    )
+
+            # Not used unless we implement other betas and alternatives to nh_p_grad
+            # if self.namelist.d_ext > 0:
+            #    raise 'Unimplemented namelist option d_ext > 0'
+            # else:
+            #    divg2 = utils.make_storage_from_shape(delz.shape,
+            #   grid.compute_origin())
```

Comment:
Are we going to do this in the next 8 months? If not I'd suggest deleting it, it won't be too bad reasoning about this through the Fortran code.

Reply:
I think if we expand the model, this has potential to be one of the first places we do so. I think as we rename everything and reorganizing, going back to the fortran code will be harder and harder. This particular one seems more important than the bigger ones like hydrostatic or grid type >=3, this is unsupported behavior on a combination of namelist options... I can uncomment the exception though and put it in the init